pegen/grammar2.py

```diff
@@ -0,0 +1,464 @@
+from __future__ import annotations  # Requires Python 3.7 or later
+
+import ast
+import re
+import sys
+import time
+import token
+import tokenize
+import traceback
+from typing import AbstractSet, Callable, Dict, Generic, Iterable, Optional, List, TypeVar, Union
+
+from pegen.parser import memoize, Parser
+from pegen.tokenizer import exact_token_types
+
+
+class GrammarNode:
+    pass
+
+
+class GrammarVisitor(object):
+
+    def visit(self, node):
+        """Visit a node."""
+        method = 'visit_' + node.__class__.__name__
+        visitor = getattr(self, method, self.generic_visit)
+        return visitor(node)
+
+    def generic_visit(self, node):
+        """Called if no explicit visitor function exists for a node."""
+        for value in node:
```

Comment:
Huh? I don't think all nodes are iterable?

Reply:
This is just a temporary hack before we set up how the nodes expose their children. I am implementing `__iter__` on all grammar nodes to obtain all subnodes. We can change this to be a descriptor or a method (node.children or similar) later.